packages/apollo-server/src/__tests__/index.test.ts

```diff
@@ -25,6 +25,35 @@ describe('apollo-server', () => {
       expect(() => new ApolloServer({ typeDefs, mocks: true })).not.toThrow;
     });
 
+    it('runs serverWillStart and serverWillStop', async () => {
+      const fn = jest.fn();
+      const oneTick = () => new Promise((res) => process.nextTick(() => res()));
```

Comment:
This is neat, TIL. What does the additional tick do for testing that isn't just awaiting a "nothing promise" a la `await new Promise(res => res())`?

Reply:
Could also shorten this slightly by doing this:

```
const oneTick = () => new Promise((res) => process.nextTick(res));
```

Reply:
Ah I forgot that Promises always resolve asynchronously! @trevor-scheer yours looks better.